rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -503,48 +502,70 @@ protected void registerPegIn(
         );
 
         PeginProcessAction peginProcessAction = peginEvaluationResult.getPeginProcessAction();
-        if (peginProcessAction == PeginProcessAction.CAN_BE_REGISTERED) {
-            logger.debug("[{}] Peg-in is valid, going to register", METHOD_NAME);
-            executePegIn(btcTx, peginInformation, totalAmount);
-        } else {
-            Optional<RejectedPeginReason> rejectedPeginReasonOptional = peginEvaluationResult.getRejectedPeginReason();
-            if (rejectedPeginReasonOptional.isEmpty()) {
-                // This flow should never be reached. There should always be a rejected pegin reason.
-                String message = "Invalid state. No rejected reason was returned from evaluatePegin method";
-                logger.error("[{}}] {}", METHOD_NAME, message);
-                throw new IllegalStateException(message);
-            }
 
-            RejectedPeginReason rejectedPeginReason = rejectedPeginReasonOptional.get();
-            logger.debug("[{}] Rejected peg-in, reason {}", METHOD_NAME, rejectedPeginReason);
-            eventLogger.logRejectedPegin(btcTx, rejectedPeginReason);
-            if (peginProcessAction == PeginProcessAction.CAN_BE_REFUNDED) {
-                logger.debug("[{}] Refunding to address {} ", METHOD_NAME, peginInformation.getBtcRefundAddress());
-                generateRejectionRelease(btcTx, peginInformation.getBtcRefundAddress(), rskTxHash, totalAmount);
+        switch (peginProcessAction) {
+            case REGISTER -> {
+                logger.debug("[{}] Peg-in is valid, going to register", METHOD_NAME);
+                executePegIn(btcTx, peginInformation, totalAmount);
+            }
+            case REFUND -> {
+                peginEvaluationResult.getRejectedPeginReason()
+                    .map(reason -> {
+                        logger.debug("[{}] Rejected peg-in, reason {}", METHOD_NAME, reason);
+                        eventLogger.logRejectedPegin(btcTx, reason);
+                        return reason;
+                    }).orElseThrow(() -> {
+                        // This flow should never be reached. There should always be a rejected pegin reason.
+                        String message = "Invalid state. No rejected reason was returned for an invalid pegin.";
+                        logger.error("[{}] {}", METHOD_NAME, message);
+                        return new IllegalStateException(message);
+                    });
+
+                logger.debug("[{}] Refunding to address {} ",
+                    METHOD_NAME, peginInformation.getBtcRefundAddress());
+                generateRejectionRelease(btcTx, peginInformation.getBtcRefundAddress(), rskTxHash,
+                    totalAmount);
```

Comment:
Looked better when all this logic was in its own method, why move it all here? It creates a code smell

Reply:
Because of the number of parameters that needs to be passed. 

```
handleRefundablePegin(BtcTransaction btcTx, Keccak256 rskTxHash,
        PeginEvaluationResult peginEvaluationResult, Address btcRefundAddress, Coin totalAmount)
```